### ISSUE TITLE
Both single and multiple message publication are blocked when using publish module

### ISSUE THREAD

**Issue by baqwas** [eclipse/paho.mqtt.python#355]
My issue is similar in behavior to the two referenced below. The outcome of the first case is unknown to me. The second case was solved through re-installation.

Using the **client** module, everything works fine. However, when using the **publish** module both functions, _single_ and _multiple_, hang. The subscriber does not receive any message. My code snippet (for the _multiple_ case) is as follows:

```
    msgs = [
        {'topic':"L1/S1/T1", 'payload':"83.0"}],
        {'topic':"L1/S1/T2", 'payload':"33.0"}
        ]
    try:
        while True:
            print("Publishing...")
            publish.multiple(msgs,
                client_id="",
                hostname="host.domain.com"),
                auth={'username':"xxxx", 'password':"xxxxxxxx"})
            print("Sleeping...")
            time.sleep(60)
    except KeyboardInterrupt:
        pass
    finally:
        pass
```
The _hostname_ and _auth_ strings have been mangled for presentation here. For the exercise, publishing repetitive values is acceptable (until I get the code working). All machines are in the same subnet.

The service broker is accessible from the publisher and the subscriber - access tested via SSH. If I revert to the **client** module use, the subscriber immediately receives messages from the client version of the publisher. Is there another set of canned examples that I can run in my subnet to leverage your **publish** module because that fits my need very nicely? Am I misunderstanding the correct use of the **publish** utility module?

Thanks a bunch.

### References:
[(https://github.com/eclipse/paho.mqtt.python/blob/m)]
[(https://github.com/eclipse/paho.mqtt.python/issues/274)]

### Installation:
```
pi@raspbari8:~ $ pip3 show paho-mqtt
Name: paho-mqtt
Version: 1.4.0
Summary: MQTT version 3.1.1 client class
Home-page: http://eclipse.org/paho
Author: Roger Light
Author-email: roger@atchoo.org
License: Eclipse Public License v1.0 / Eclipse Distribution License v1.0
Location: /home/pi/.local/lib/python3.5/site-packages
Requires: 
pi@raspbari8:~ $ 
```

**Comment by nickma82** [comment]
I'm also running into a similar issue with paho-mqtt v1.4.0.
In my case publish.single (not tested with multiple but as single is a wrapper of multiple.. ) is working when the connection and auth credentials are ok, otherwise it loops here infinitely:
https://github.com/eclipse/paho.mqtt.python/blob/a8b2b4d039571e9bd873dd550a1dddabc6d3cb61/src/paho/mqtt/client.py#L2967-L2972

**Comment by baqwas** [comment]
@nickma82 
I think I will follow your lead now and _split_ the multiple payload into serial singles. Of course, I will have to fall back on the **client** and not the **publish** module. Thanks again for posting the note.

Regards,

**Comment by nickma82** [comment]
Hi @baqwas, 
thanks for the quick response!

> split the multiple payload into serial singles

Isn't it possible to replace the `loop_forever` call, with a `loop(timeout, ...)` like call?
https://github.com/eclipse/paho.mqtt.python/blob/a8b2b4d039571e9bd873dd550a1dddabc6d3cb61/src/paho/mqtt/publish.py#L160

Workaround:
[This PR #373 ](https://github.com/eclipse/paho.mqtt.python/pull/373) enables a pretty neat workaround.
By calling `publish` "manually" followed by an `wait_for_publish(timeout)` everything works just fine.

```
infot = self.mqttc.publish(topic, msg, qos=1)
infot.wait_for_publish(0.5)
```

**Comment by ralight** [comment]
I don't think this should be closed, the problem with invalid authentication blocking the call should be fixed.

**Comment by nickma82** [comment]
@ralight I'd rather see a good fix too @baqwas.

**Comment by MattBrittan** [comment]
I'm going to close this because it appears the issue has been addressed (as noted in issue #470) and the last activity was back in 2019. 

Note: This is part of an exercise to clean up old issues so that the project can move forwards. Due to the number of issues being worked through mistakes will be made; please feel free to reopen this issue (or comment) if you believe it's been closed in error.
